### ISSUE TITLE
Inconsistency on tabnet between paper and code

### ISSUE THREAD

**Issue by titobrezende** [google-research/google-research#717]
Hi

For me there are two items on [https://github.com/google-research/google-research/blob/master/tabnet/tabnet_model.py](url)  that are different from what was reported in the paper [https://arxiv.org/pdf/1908.07442.pdf](url) 

On **interpretability** section the author @soarik  says that:
<img src="https://render.githubusercontent.com/render/math?math=\sum_{c=1}^{N_d} ReLU(d_{b,c}[i])">
But the given code in github is dividing this sum by the number of steps:
(line 177)
`# Aggregated masks are used for visualization of the`
` # feature importance attributes.`
 `         scale_agg = tf.reduce_sum(
              decision_out, axis=1, keep_dims=True) / (
                  self.num_decision_steps - 1)`
Also, for the aggregate feature importance mask, it looks like the normalization is missing. The paper states:
<img src="https://render.githubusercontent.com/render/math?math=\frac{ \sum_{i=1}^{N_{steps}} \eta_b[i]M_{b,j}[i]} {\sum_{j=1}^{D}\sum_{i=1}^{N_{steps}} \eta_b[i]M_{b,j}[i]}">
But the give code is:
(line 182)
`aggregated_mask_values += mask_values * scale_agg`
Inside a for looping over each step, which will give only the top part of the equation.

Am I missing something or the paper and code are really different?

**Comment by yhkll** [comment]
> Hi
> 
> For me there are two items on [https://github.com/google-research/google-research/blob/master/tabnet/tabnet_model.py](url) that are different from what was reported in the paper [https://arxiv.org/pdf/1908.07442.pdf](url)
> 
> On **interpretability** section the author @soarik says that: ![](https://render.githubusercontent.com/render/math?math=\sum_{c=1}^{N_d} ReLU(d_{b,c}[i])) But the given code in github is dividing this sum by the number of steps: (line 177) `# Aggregated masks are used for visualization of the` ` # feature importance attributes.` ` scale_agg = tf.reduce_sum( decision_out, axis=1, keep_dims=True) / ( self.num_decision_steps - 1)` Also, for the aggregate feature importance mask, it looks like the normalization is missing. The paper states: ![](https://render.githubusercontent.com/render/math?math=\frac{ \sum_{i=1}^{N_{steps}} \eta_b[i]M_{b,j}[i]} {\sum_{j=1}^{D}\sum_{i=1}^{N_{steps}} \eta_b[i]M_{b,j}[i]}) But the give code is: (line 182) `aggregated_mask_values += mask_values * scale_agg` Inside a for looping over each step, which will give only the top part of the equation.
> 
> Am I missing something or the paper and code are really different?

Hello,as the README.md said,I use `python -m test_experiment_covertype.py` for simple test,but its result is obviously wrong,The traing loss and val accuracy are irregular,just because it's  in the low-resource environment?How can I fix this problem?
![KH HVLC(7N@1KB$A@Y%FBDF](https://user-images.githubusercontent.com/41032255/139565232-955f7e1f-b1e8-41bf-b6ac-4df88770aa7b.png)
